application/frontend/controllers/MfaController.php

```diff
@@ -214,6 +215,41 @@ public function actionDelete(int $id)
         return null;
     }
 
+    /**
+     * Delete one u2f/webauthn credential from an MFA record
+     * @param int $mfaId
+     * @param string $credId
+     * @return null
+     * @throws \Throwable
+     * @throws BadRequestHttpException
+     * @throws NotFoundHttpException
+     * @throws ServerErrorHttpException
+     */
+    public function actionDeleteCredential(int $mfaId, string $credId)
+    {
+        $mfa = $this->getRequestedMfa($mfaId);
+
+        if ($mfa->type != Mfa::TYPE_WEBAUTHN) {
```

Comment:
I might throw this error in `deleteCredential` instead of returning true. Then delete this if statement.